### ISSUE TITLE
Problem of the loss when training from scratch

### ISSUE THREAD

**Issue by melohux** [eladrich/pixel2style2pixel#113]
Hello, thank you for your work! But I've met some loss problem when training from scratch: it will have nan loss and I cannot generate images from latent code. Do you have any idea?

> $ python scripts/train.py --dataset_type=ffhq_encode --exp_dir=./path/to/experiment --workers=8 --batch_size=8 --test_batch_size=8 --test_workers=8 --val_interval=2500 --save_interval=5000 --encoder_type=GradualStyleEncoder --start_from_latent_avg --lpips_lambda=0.8 --l2_lambda=1 --id_lambda=0.1
./path/to/experiment
{'batch_size': 8,
 'board_interval': 1,
 'checkpoint_path': None,
 'dataset_type': 'ffhq_encode',
 'encoder_type': 'GradualStyleEncoder',
 'exp_dir': './path/to/experiment',
 'id_lambda': 0.1,
 'image_interval': 100,
 'input_nc': 3,
 'l2_lambda': 1.0,
 'l2_lambda_crop': 0,
 'label_nc': 0,
 'learn_in_w': False,
 'learning_rate': 0.0001,
 'lpips_lambda': 0.8,
 'lpips_lambda_crop': 0,
 'max_steps': 500000,
 'optim_name': 'ranger',
 'resize_factors': None,
 'save_interval': 5000,
 'start_from_latent_avg': True,
 'stylegan_weights': '/mnt/sdb/xiwei/model/stylegan2-ffhq-config-f.pt',
 'test_batch_size': 8,
 'test_workers': 8,
 'train_decoder': False,
 'val_interval': 2500,
 'w_norm_lambda': 0,
 'workers': 8}
Loading encoders weights from irse50!
Loading decoder weights from pretrained!
Loading ResNet ArcFace
Loading dataset for ffhq_encode
Number of training samples: 69000
Number of test samples: 3897
Metrics for train, step 0
        loss_id =  1.0162537097930908
        id_improve =  -1.0162536224815995
        loss_l2 =  0.25566887855529785
        loss_lpips =  0.8953849077224731
        loss =  1.0736021995544434
Metrics for train, step 1
        loss_id =  1.0107078552246094
        id_improve =  -1.0107079751323909
        loss_l2 =  0.2594793438911438
        loss_lpips =  0.9074615240097046
        loss =  1.0865193605422974
Metrics for train, step 2
        loss_id =  nan
        id_improve =  nan
        loss_l2 =  inf
        loss_lpips =  0.7421762943267822
        loss =  nan
Metrics for train, step 3
        loss_id =  nan
        id_improve =  nan
        loss_l2 =  inf
        loss_lpips =  0.7516516447067261
        loss =  nan

And the generated images are always noise. If I import checkpoint of psp, I still have the same problem of loss. However, the inference with given checkpoint outputs normal results.

**Comment by melohux** [comment]
I find that it is because that I've changed the line 28 in `coach.py` as 
> 		self.device = 'cuda:1'  
When I change it back as 'cuda:0', it will be fine.
